### ISSUE TITLE
rust ALEGoToDefinition does not trigger any action

### ISSUE THREAD

**Issue by milosgajdos** [dense-analysis/ale#4750]
## Information

**VIM version**

```
VIM - Vi IMproved 9.1 (2024 Jan 02, compiled Mar 31 2024 17:57:16)
macOS version - arm64
```

Operating System: 

```
macOS Sonoma 14.4.1 (23E224)
Darwin straum 23.4.0 Darwin Kernel Version 23.4.0: Fri Mar 15 00:12:49 PDT 2024; root:xnu-10063.101.17~1/RELEASE_ARM64_T6020 arm64
```

## What went wrong

It would appear that `rust-analyzer` stopped working as expected at some point. I'm coming back to `rust` after a while and realized whenever I need to navigate to a specific func (i.e. calling `ALEGoToDefinition`) the action does not execute at all.

The same goes for splitting the definitions i.e. running `:ALEGoToDefinition -split`. No action gets traiggered.

## Reproducing the bug

Here's a simple rust program
```rust
use tokio::io::AsyncWriteExt;
use tokio::net::{TcpListener, TcpStream};

#[tokio::main]
async fn main() {
    let listener = TcpListener::bind("127.0.0.1:8181").await.unwrap();

    loop {
        let (stream, _) = listener.accept().await.unwrap();
        handle_connection(stream).await;
    }
}

async fn handle_connection(mut stream: TcpStream) {
    let contents = "{\"foo\": \"bar\"}";

    let response = format!(
        "HTTP/1.1 200 OK\r\nContent-Type: application/json\r\nContent-Length: {}\r\n\r\n{}",
        contents.len(),
        contents
    );
    stream.write(response.as_bytes()).await.unwrap();
    stream.flush().await.unwrap();
}

```

1. I did this.

Move your cursor to `accept` in ` listener.accept().await.unwrap();`. Do either `ALEGoToDefinition` or `ALEGoToDefinition -split`

3. Then this happened.

Nothing happens. No action is triggered.

### :ALEInfo
<details>
  <summary>Expand</summary>

```

 Current Filetype: rust
Available Linters: ['analyzer', 'cargo', 'cspell', 'rls', 'rustc']
   Linter Aliases:
'analyzer' -> ['rust_analyzer']
  Enabled Linters: ['analyzer']
  Ignored Linters: []
 Suggested Fixers:
  'remove_trailing_lines' - Remove all blank lines at the end of a file.
  'rustfmt' - Fix Rust files with Rustfmt.
  'trim_whitespace' - Remove all trailing whitespace characters at the end of every line.
 
 Linter Variables:
let g:ale_rust_analyzer_config = {}
let g:ale_rust_analyzer_executable = 'rust-analyzer'
let g:ale_rust_rustfmt_executable = 'rustfmt'
let g:ale_rust_rustfmt_options = ''
 
 Global Variables:
let g:ale_cache_executable_check_failures = v:null
let g:ale_change_sign_column_color = v:null
let g:ale_command_wrapper = ''
let g:ale_completion_delay = 100
let g:ale_completion_enabled = 1
let g:ale_completion_max_suggestions = 50
let g:ale_disable_lsp = 'auto'
let g:ale_echo_cursor = 1
let g:ale_echo_msg_error_str = 'Error'
let g:ale_echo_msg_format = '[%linter%] %s [%severity%]'
let g:ale_echo_msg_info_str = 'Info'
let g:ale_echo_msg_warning_str = 'Warning'
let g:ale_enabled = 1
let g:ale_fix_on_save = 1
let g:ale_fixers = {'rust': ['rustfmt', 'trim_whitespace', 'remove_trailing_lines'], 'python': ['ruff', 'ruff_format']}
let g:ale_history_enabled = 1
let g:ale_info_default_mode = 'preview'
let g:ale_history_log_output = 1
let g:ale_keep_list_window_open = v:null
let g:ale_lint_delay = 200
let g:ale_lint_on_enter = 1
let g:ale_lint_on_filetype_changed = 1
let g:ale_lint_on_insert_leave = 1
let g:ale_lint_on_save = 1
let g:ale_lint_on_text_changed = 'normal'
let g:ale_linter_aliases = {}
let g:ale_linters = {'rust': ['analyzer'], 'python': ['ruff', 'pylsp']}
let g:ale_linters_explicit = 0
let g:ale_linters_ignore = {}
let g:ale_list_vertical = v:null
let g:ale_list_window_size = v:null
let g:ale_loclist_msg_format = v:null
let g:ale_max_buffer_history_size = 20
let g:ale_max_signs = v:null
let g:ale_maximum_file_size = v:null
let g:ale_open_list = v:null
let g:ale_pattern_options = v:null
let g:ale_pattern_options_enabled = v:null
let g:ale_root = {}
let g:ale_set_balloons = 0
let g:ale_set_highlights = 1
let g:ale_set_loclist = 1
let g:ale_set_quickfix = 0
let g:ale_set_signs = 1
let g:ale_sign_column_always = 1
let g:ale_sign_error = '✘'
let g:ale_sign_info = v:null
let g:ale_sign_offset = v:null
let g:ale_sign_style_error = v:null
let g:ale_sign_style_warning = v:null
let g:ale_sign_warning = '⚠'
let g:ale_sign_highlight_linenrs = v:null
let g:ale_type_map = {}
let g:ale_use_neovim_diagnostics_api = 0
let g:ale_use_global_executables = v:null
let g:ale_virtualtext_cursor = 'all'
let g:ale_warn_about_trailing_blank_lines = 1
let g:ale_warn_about_trailing_whitespace = 1
 
  Command History:

(executable check - success) rust-analyzer
(finished - exit code 1) ['/bin/zsh', '-c', '''rustfmt'' < ''/var/folders/8k/x072lv2x0g73g7mqtchl5xpm0000gn/T/vX8kzEe/54/main.rs''']
```
</details>

I have the following `rust` related config

### Vim Config
<details>
  <summary>Expand</summary>

```
" rust.vim settings
" fmt code on autosave
let g:rustfmt_autosave = 1
let g:rustfmt_emit_files = 1
let g:rust_clip_command = 'pbcopy'
let g:rust_cargo_use_clippy = 1
let g:ale_linters.rust = ['analyzer']
let g:ale_fixers.rust = ['rustfmt', 'trim_whitespace', 'remove_trailing_lines']
set hidden
" jump to definitions
au FileType rust nmap <Leader>g <Plug>(ale_go_to_definition)
au FileType rust nmap <leader>y <Plug>(ale_find_references)
au FileType rust nmap <Leader>s <Plug>(ale_go_to_definition_in_split)
au FileType rust nmap <Leader>v <Plug>(ale_go_to_definition_in_vsplit)
au FileType rust nmap <Leader>e <Plug>(ale_rename)
au FileType rust nmap <Leader>f <Plug>(ale_fix)
au FileType rust nmap <Leader>h <Plug>(ale_hover)
" close ALE hover preview
au FileType rust nnoremap <Leader>a :pclose<CR>

au FileType rust nmap <silent> <C-n> <Plug>(ale_next_wrap)
au FileType rust nmap <silent> <C-m> <Plug>(ale_previous_wrap)
```
</details>

**Comment by milosgajdos** [comment]
Closing as this was due to `rust-analyser` not being installed 🫠

As it turns out `rustup` has a `rust-analyzer` _SHIM_ which I assumed was an actual binary. Alas, no.

Apologies about the noise, hope this might help someone:

```
rustup component add rust-analyzer
```
